hummingbot/market/market_base.pyx

```diff
@@ -389,7 +389,7 @@ cdef class MarketBase(NetworkIterator):
         return self.c_quantize_order_amount(trading_pair, amount)
 
     def supported_order_types(self):
-        return [OrderType.LIMIT]
```

Comment:
@Nullably 
I notice you changed this to just return OrderType.LIMIT in a previous PR.
I changed it to return both OrderType.LIMIT and OrderType.MARKET.
The reason is that returning just OrderType.LIMIT causes DEXs or PaperTrade to misbehave in XEMM because it can't differentiate maker from taker. But adding OrderType.MARKET makes it use LIMIT as maker and MARKET as taker for those connectors. Let me know if you think this change would break any part of the code so we can agree on a lasting solution for all cases.

Reply:
since you're making PR for the hummingsim, i think it's best this function raises NotImplemented in the base class, s.t. all sub classes need to implement this, including the `market` class in the hummingsim. And there, you can return all the 3 types of the order.

Reply:
The main reason that I removed the LIMIT_MAKER from this is to get tests passed, hummingsim doesn't understand LIMIT_MAKER, it will just raise error.